nixos/modules/services/networking/zeronet.nix

```diff
@@ -0,0 +1,527 @@
+          type = types.nullOr types.str;
+          default = null;
+          description = ''
+          '';
+        };
+
+        fixFloatDecimals = mkOption {
+          type = types.nullOr types.str;
+          default = null;
+          description = ''
+          '';
+        };
+
+        dbMode = mkOption {
+          type = types.nullOr types.str;
+          default = null;
+          description = ''
+          '';
+        };
+
+        downloadOptional = mkOption {
+          type = types.nullOr types.str;
+          default = null;
+          description = ''
+          '';
+        };
+
+        coffeescriptCompiler = mkOption {
+          type = types.nullOr types.str;
+          default = null;
+          description = ''
+          '';
+        };
+
+        tor = mkOption {
+          type = types.nullOr types.str;
+          default = null;
+          description = ''
+          '';
+        };
+
+        torController = mkOption {
+          type = types.nullOr types.str;
+          default = null;
+          description = ''
+          '';
+        };
+
+        torProxy = mkOption {
+          type = types.nullOr types.str;
+          default = null;
+          description = ''
+          '';
+        };
+
+        torPassword = mkOption {
+          type = types.nullOr types.str;
+          default = null;
+          description = ''
+          '';
+        };
+
+        torHsLimit = mkOption {
+          type = types.nullOr types.str;
+          default = null;
+          description = ''
+          '';
+        };
+
+        bitResolver = mkOption {
+          type = types.nullOr types.str;
+          default = null;
+          description = ''
+          '';
+        };
+
+        optionalLimit = mkOption {
+          type = types.nullOr types.str;
+          default = null;
+          description = ''
+          '';
+        };
+
+        pinBigfile = mkOption {
+          type = types.nullOr types.str;
+          default = null;
+          description = ''
+          '';
+        };
+
+        autodownloadBigfileSizeLimit = mkOption {
+          type = types.nullOr types.str;
+          default = null;
+          description = ''
+          '';
+        };
+
+        extraConfig = mkOption {
+          type = types.lines;
+          default = "";
+          description = ''
+          Extra configuration to be appended to the zeronet config file
+          '';
+        };
+      };
+    };
+
+    config = mkIf (cfg.enable) {
+      users.users."${cfg.user}" = if cfg.user == "zeronet" then {
+        isSystemUser = true;
+        group = cfg.group;
+        home = cfg.dataDir;
+        createHome = true;
+      } else {};
+
+      users.groups."${cfg.group}" = {};
+
+      systemd.services.zeronet = {
+        enable = true;
+        after = [ "network.target" ];
+        wantedBy = [ "multi-user.target" ];
+        serviceConfig.ExecStart  = "${pkgs.zeronet}/bin/zeronet.py --conf_file ${configFile}";
```

Comment:
Executables should not have an extension, and especially in the case of Python where it would create bytecode.